### ISSUE TITLE
SCIM: Require search filter for GET

### ISSUE THREAD

**Issue by nynymike** [GluuFederation/oxTrust#317]
We do not want a SCIM client to ever send an search to `ou=people,o=(org-inum),o=gluu` for all users. 
Can we prevent this by requiring some filter, or returning an error?

```
access.20160803061532Z:[03/Aug/2016:05:28:23 +0000] SEARCH REQ conn=6 op=1329 msgID=1330 base="ou=people,o=@!DF52.8A2B.1610.1D59!0001!F186.A7D4,o=gluu" scope=sub filter="(&(objectClass=top)(objectClass=gluuPerson))" attrs=""
access.20160803061532Z:[03/Aug/2016:05:28:23 +0000] SEARCH RES conn=6 op=1329 msgID=1330 result=0 nentries=100 unindexed etime=10517581
```

**Comment by uboznikov** [comment]
In SCIM 2.0, an error is returned:

[MAX_COUNT is 200](https://github.com/GluuFederation/oxTrust/blob/master/server/src/main/java/org/gluu/oxtrust/ws/rs/scim2/UserWebService.java#L90-L93)

Also, the incorrect POST Search has been deprecated and will be fixed in v2.4.5.

[SCIM-Client personSearch](https://github.com/GluuFederation/SCIM-Client/blob/master/src/main/java/gluu/scim2/client/BaseScim2ClientImpl.java#L964-L965)
[oxTrust personSearch](https://github.com/GluuFederation/oxTrust/blob/master/server/src/main/java/org/gluu/oxtrust/ws/rs/scim2/UserWebService.java#L424)

I'm still checking SCIM 1.1.

**Comment by uboznikov** [comment]
The SCIM 1.1 search will be updated and will essentially be the same as SCIM 2.0.

[SCIM 1.1 Protocol 3.2.2.1. Filtering](https://tools.ietf.org/html/draft-scim-api-01#section-3.2.2.1)
